code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -266,4 +181,293 @@
+			if(reagents.total_volume < reagents.maximum_volume) // don't recharge reagents and drain power if the storage is full
+				robot.cell.use(synth_cost)
+				..()
+
+/obj/item/weapon/reagent_containers/glass/replenishing/cyborg/hacked
+	name = "mickey finn's special brew"
+	reagent_list = BEER2
+	synth_cost = 25 //4165 cell charge for 50u !NotShitterJuice.
+
+//Grippers: Simple cyborg manipulator. Limited use... SLIPPERY SLOPE POWERCREEP
+/obj/item/weapon/gripper
+	icon = 'icons/obj/device.dmi'
+	actions_types = list(/datum/action/item_action/magrip_drop)
+	var/obj/item/wrapped = null // Item currently being held.
+	var/list/can_hold = list() //Has a list of items that it can hold.
+	var/list/blacklist = list() //This is a list of items that can't be held even if their parent is whitelisted.
+	var/force_holder = null
+
+/datum/action/item_action/magrip_drop
+	name = "Drop Item"
+
+/datum/action/item_action/magrip_drop/Trigger()
+	var/obj/item/weapon/gripper/G = target
+	if(!istype(G))
+		return
+	G.drop_item(force_drop = 1)
+
+/obj/item/weapon/gripper/proc/grip_item(obj/item/I as obj, mob/user, var/feedback = TRUE)
+	//This function returns TRUE if we successfully took the item, or FALSE if it was invalid. This information is useful to the caller
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(I.loc == (R || R.module))// Don't remove your own modules
+			to_chat(R, "<span class='danger'>ERROR. Safety protocols prevents self-disassembling.</span>")
+			return FALSE
+	if (!wrapped)
+		if(is_type_in_list(I, can_hold) && !is_type_in_list(I, blacklist))
+			if(feedback)
+				to_chat(user, "<span class='notice'>You collect \the [I].</span>")
+			I.loc = src
+			I.add_fingerprint(user)
+			wrapped = I
+			update_icon()
+			return TRUE
+		if(feedback)
+			to_chat(user, "<span class='danger'>ERROR. Your [src.name] wasn't designed to handle \the [I].</span>")
+		return FALSE
+	if(feedback)
+		to_chat(user, "<span class='danger'>ERROR. Your [src.name] is already holding \the [wrapped].</span>")
+	return FALSE
+
+
+/obj/item/weapon/gripper/proc/drop_item(var/obj/item/to_drop, var/atom/target, force_drop = 0,var/dontsay = null) //Do we care about to_drop at all?
+	if(!gripper_sanity_check(src))
+		return FALSE
+	if(to_drop && !istype(wrapped, to_drop))// What the fuck?
+		drop_item(force_drop = 1)
+		return FALSE
+	if(!target) //Just drop it, baka.
+		target = src.loc
+	if(!dontsay)
+		to_chat(usr, "<span class='warning'>You drop \the [wrapped].</span>")
+	wrapped.dropped(usr)
+	if(force_drop)
+		wrapped.loc = get_turf(target)
+	else
+		wrapped.forceMove(target)
+	wrapped = null
+	update_icon()
+	return TRUE
+
+/obj/item/weapon/gripper/proc/gripper_safety_check(var/mob/user, var/target)
+	if(issilicon(user))
+		var/mob/living/silicon/robot/A = user
+		if(!A.emagged)
+			to_chat(user, "<span class='danger'>ERROR. Safety protocols prevent your [src.name] from [ismob(target)? "completing this action." : "holding \the [target]."]</span>")
```

Comment:
Have you verified that this `?` works? There is no space between it and the `ismob(target)`.